src/core/services/interimElement/interimElement.js

```diff
@@ -526,20 +526,19 @@ function InterimElementProvider() {
 
           if ( options.$destroy === true ) {
 
-            return hideElement(options.element, options).then(function(){
-              (isCancelled && rejectAll(response)) || resolveAll(response);
-            });
+            return hideElement(options.element, options).then(hideSuccessful);
 
           } else {
             $q.when(showAction).finally(function() {
-              hideElement(options.element, options).then(function() {
-                isCancelled ? rejectAll(response) : resolveAll(response);
-              }, rejectAll);
+              hideElement(hideSuccessful, rejectAll);
             });
 
             return self.deferred.promise;
           }
 
+          function hideSuccessful(hideResult){
+              (isCancelled && rejectAll(hideResult)) || resolveAll(hideResult);            
```

Comment:
Isn't the `hideResult` different from the previous one? Previously it always passed over the provided `response`.